code/datums/extensions/scent/_scent.dm

```diff
@@ -63,16 +63,30 @@ Scent intensity
 		return PROCESS_KILL
 	emit_scent()
 
+/datum/extension/scent/proc/check_smeller(var/mob/living/carbon/human/smeller)
+	if(!istype(smeller))
+		return FALSE
+	if(smeller.stat != CONSCIOUS)
+		return FALSE
+	if(smeller.failed_last_breath || smeller.wear_mask)
+		return FALSE
+	if(smeller.head && smeller.head.permeability_coefficient < 1)
```

Comment:
Since there is work being done on the inventory system, you may want to consider using the various inventory procs like `get_equipped_item(SLOT)`

Reply:
I will need to change those lines again anyway since the new inventory code replaces slot definitions.